bin/manage.py

```diff
@@ -0,0 +1,406 @@
+    """
+    val = os.environ.get(key, default)
+    try:
+        val = val.split('#')[0]
+        val = val.strip()
+    finally:
+        # just swallow AttributeErrors for non-strings
+        return val
+
+# ---------------------------------------------------------
+
+SESSION_CACHE_FILE = get_environ('SESSION_CACHE_FILE', '/tmp/mongodb-session')
+SESSION_NAME = get_environ('SESSION_NAME', 'mongodb-replica-set-lock')
+SESSION_TTL = int(get_environ('SESSION_TTL', 60))
+
+# consts for node state
+PRIMARY = 'mongodb-primary'
+SECONDARY = 'mongodb-secondary'
+REPLICA = 'mongodb'
+
+# key where primary will be stored in consul
+PRIMARY_KEY = get_environ('PRIMARY_KEY', 'mongodb-primary')
+
+# how many time to retry connecting to mongo
+# be aware that the health check may timeout before this is reached
+MONGO_RETRY_TIMES=int(get_environ('MONGO_RETRY_TIMES', 10))
+
+# ---------------------------------------------------------
+
+class ContainerPilot(object):
+    """
+    ContainerPilot config is where we rewrite ContainerPilot's own config
+    so that we can dynamically alter what service we advertise
+    """
+
+    def __init__(self):
+        # TODO: we should make sure we can support JSON-in-env-var
+        # the same as ContainerPilot itself
+        self.path = get_environ('CONTAINERPILOT', None).replace('file://', '')
+        with open(self.path, 'r') as f:
+            self.config = json.loads(f.read())
+
+    @debug
+    def update(self, state):
+        if state and self.config['services'][0]['name'] != state:
+            self.config['services'][0]['name'] = state
+            self.render()
+            return True
+
+    @debug
+    def render(self):
+        new_config = json.dumps(self.config)
+        with open(self.path, 'w') as f:
+            f.write(new_config)
+
+    def reload(self):
+        """ force ContainerPilot to reload its configuration """
+        log.info('Reloading ContainerPilot configuration.')
+        os.kill(1, signal.SIGHUP)
+
+# ---------------------------------------------------------
+# Top-level functions called by ContainerPilot or forked by this program
+
+@debug
+def pre_start():
+    """
+    MongoDB must be running in order to execute most of our setup behavior
+    so we're just going to make sure the directory structures are in
+    place and then let the first health check handler take it from there
+    """
+    # TODO is there anything that needs to be done before starting mongo?
+    sys.exit(0)
+
+@debug
+def health():
+    """
+    Run a simple health check. Also acts as a check for whether the
+    ContainerPilot configuration needs to be reloaded (if it's been
+    changed externally).
+    """
+    # TODO periodic mongodumps to Manta
+
+    hostname = socket.gethostname()
+    ip = get_ip()
+    local_mongo = MongoClient(ip, connect=False)
+
+    #  - check that mongo is responsive
+    if not is_mongo_up(local_mongo):
+        return False
```

Comment:
It should; I modified the caller: https://github.com/autopilotpattern/mongodb/pull/1/files#diff-70dd0315fb7fe82d946cf97303e35b87R414